overlord/devicestate/firstboot.go

```diff
@@ -170,6 +170,10 @@ func populateStateFromSeedImpl(st *state.State, opts *populateStateFromSeedOptio
 		if beginTask != nil {
 			// hooks must wait for mark-preseeded
 			hooksTask.WaitFor(preseedDoneTask)
+			if n := len(all); n > 0 {
```

Comment:
AIUI this will serialize the install of the snaps. Could we also do something like:
```
                if beginTask != nil {
                        // hooks must wait for mark-preseeded
                        hooksTask.WaitFor(preseedDoneTask)
+                       markSeeded.WaitFor(preseedDoneTask)
                        if lastBeforeHooksTask != nil {
                                beginTask.WaitFor(lastBeforeHooksTask)
                        }
```
instead which would not serialize things? Or is that risky/hard to do with the exiting way we do stuff?

Reply:
Did you mean `markSeeded.WaitFor(hooksTask)`? markSeeded already waits for preseedDoneTask, it's done at the bottom of the function (doesn't need to be done here where we're looping over snaps).

We could certainly think about letting this go in parallel here although I'm not 100% sure this wins anything due to state locking within the tasks.

Reply:
As discussed in the standup for now we leave the fix as is and run things sequentially. Let's just keep in mind that this might be optimization potential in the future (but it's unclear how much given that we usually don't have many snaps, even less hooks and that hooks are generally fast).